dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -163,35 +162,29 @@ internal KernelFunction(string name, string? pluginName, string description, IRe
 
         TagList tags = new() { { MeasurementFunctionTagName, this.Name } };
         long startingTimestamp = Stopwatch.GetTimestamp();
-        FunctionResult? functionResult = null;
+        FunctionResult? functionResult = new(this);
```

Comment:
Why is this initialized to something other than null? In what situation do we get to the return without this having been written to?

Reply:
Currently we initialize it as null, but in fact, `InvokeAsync` in `KernelFunction` never returns null, the result type is also non-nullable, it's either some result or exception. With exception handling scenarios, I thought it would be better to initialize `FunctionResult` with a link to `KernelFunction`, so user will be able to understand which function failed, with with `null` result value. Otherwise, we need to think how we can tell user an info about failed function, maybe put it in specific exception type?

Reply:
Based on latest changes, this was reverted to be initialized as null.

Reply:
FYI: With latest changes, it's again initialized as non-nullable. This is to make `FunctionResult` property inside filter context as non-nullable. In this case, users won't be forced to check whether `context.Result` is null every time, which simplifies the usage. Before filter invocation, `context.Result` will contain information about Function and kernel Culture, with `null` Value. After invocation, it will be the same, but Value will hold result of function, and Metadata dictionary will be populated.